lib/enumerable/with_delayed_exceptions.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def with_delayed_exceptions(*catch, &block)
```

Comment:
Not sure what the common convention is, but I'm strongly tempted to scope this into our own module and include it in `TemplateSets` instead of monkey patching `Enumerable`. (Or is this already scoped to the k8s-deploy package?)

Reply:
This pollutes the global `Enumerable` module in ruby. So anyone requiring it or something that requires it will get it.  

I think you're right and I'll look into scoping this down, though I suspect the signature will change (e.g. you pass in the list to iterate over)  since the current receiver is anything that includes `enumerable` not a TemplateSet(s).